README.md

```diff
@@ -222,6 +222,187 @@ ok
+
+{ patient_user_requests, [
+    { max_time, 10000}, 
+    { regulators, [{rate, [ { limit, 1000 } ] } 
+  ] 
+}
+
+```
+* Let's create **_impatient_user_requests_** queue that will keep requests in the queue for up to 200 milliseconds.  
+Additionally, we'll make it a LIFO queue. Unfair, but if we assume that happy/unhappy is a boolean  
+we're likely to maximize the happy users!
+ 
+     
+```erlang
+ 
+{ impatient_user_requests, [
+    { max_time, 200}, 
+    { type, lifo},
+    { regulators, [{rate, [ { limit, 1000 } ] } 
+  ]
+}
+
+```
+   
+NOTE: In order to pace requests from both queues at 1000 per second, use **group_rate** regulation (EXAMPLE 4)
+  
+  
+####EXAMPLE 4: 
+* Rate regulate http requests from multiple queues
+
+Create **group_rates** regulator called **_http_request_rate_**  and assign it to both _impatient_user_requests_ and _patient_user_requests_ 
+                                                                           
+```erlang
+
+{ jobs, [ 
+    { group_rates,[{ http_request_rate, [{limit,1000}] }] },     
+    { queues, [    
+        { impatient_user_requests,  
+            [ {max_time, 200}, {type, lifo}, {regulators,[{ group_rate, http_request_rate}]} ] },    
+        { patient_user_requests, 
+            [ {max_time, 10000}, {regulators,[{ group_rate, http_request_rate} ]} 
+      ]
+    } 
+  ]  
+}
+
+```      
+
+####EXAMPLE 5: 
+* Can't afford to drop http requests on the floor once max_size is reached?
+* Implement and use your own queue to persist those unfortunate http requests and serve them eventually
+  
+```erlang
+
+ -module(my_persistent_queue).  
+ -behaviour(jobs_queue).  
+ -export([  new/2,  
+            delete/1,          
+            in/3,  
+            out/2,  
+            peek/1,  
+            info/2,  
+            all/1]). 
+             
+ ## implementation  
+ ...
+ 
+```
+ 
+Configuration:
+ 
+```erlang
+
+{ jobs, [  
+    { queues, [  
+        { http_requests, [
+            { mod, my_persistent_queue},  
+            { max_size, 10000 },  
+            { regulators, [ { rate, [ { limit, 1000 } ] } ] } 
+          ]
+        }  
+      ]
+    }  
+  ] 
+} 
+
+```
+ 
+ 
+###The use of sampler framework
+1. Get a sampler running and sending feedback to the jobs server.
+2. Apply its feedback to a regulator limit. 
+
+####EXAMPLE 6: 
+* Adjust rate regulator limit on the fly based on the feedback from **jobs_sampler_cpu** named **_cpu_feedback_**
+
+```erlang
+
+{ jobs, [  
+    { samplers, [{ cpu_feedback, jobs_sampler_cpu, [] } ] },  
+    { queues, [ 
+        { http_requests, [ 
+            { regulators,   [ { rate, [ { limit,1000 } ] }, 
+            { modifiers,    [ { cpu_feedback,  10} ] }  
+          ]
+        }  
+      ] 
+    }  
+  ]
+}  
+
+```
+ NOTE 1: 10 = Percent increment by which to modify the regulator limit
+ 
+ NOTE 2: In this example, when CPU usage increases by %27, the sampler will send 70, the percentage to be applied to the predefined limit.  
```

Comment:
Well, actually, if the jobs_sampler_cpu sampler is defined with no arguments, it will use the default `{levels, [{80,1},{90,2},{100,3}]}`. The algorithm tries to calculate some "average" per-cpu load, so 100 would be max. But given this setting, an average per-cpu load of 85% would give a sampler feedback of `1`. That is, the `levels` algorithm (or `jobs_sampler:calc(value, Levels, History)` will pick the highest level where `Value >= Level` - in this case 80.